app/controllers/api/v1/users_controller.rb

```diff
@@ -27,6 +27,19 @@ def show
       end
     end
 
+    def search
+      if params[:q].present?
+        users = User.where("name ~* ?", params[:q])
+      else
+        users = User.none
+      end
+      users = users.by_email
```

Comment:
Yes but that's only because now we are doing an exact match.
Later on it will be a fuzzy match and we shouldn't be changing the API contract around the ordering of results